hoomd/hpmc/IntegratorHPMCMonoNEC.h

```diff
@@ -0,0 +1,1275 @@
+                                        // calculate delta_pos on touch
+                                        collisionPlaneVector = r_ij - direction * newDist ;
+                                        }
+                                    else
+                                        {
+                                        if( newDist == sweepableDistance )
+                                            {
+                                            this->m_exec_conf->msg->error() << "Two particles with the same distance" << std::endl;
+                                            }
+                                        }
+                                    }
+                                else
+                                    {
+                                    this->m_exec_conf->msg->error() << "Two particles overlapping [with negative sweepable distance]." << i << " and " << j  << std::endl;
+                                    this->m_exec_conf->msg->error() << "Proceeding with the new particle without moving the initial one" << std::endl;
+
+                                    sweepableDistance = 0.0;
+                                    next = j;
+                                    collisionPlaneVector = r_ij;
+
+                                    return sweepableDistance;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            else
+                {
+                // skip ahead
+                cur_node_idx += this->m_aabb_tree.getNodeSkip(cur_node_idx);
+                }
+            }  // end loop over AABB nodes
+        } // end loop over images
+
+    return sweepableDistance;
+    }
+
+
+
+
+// sweepableDistance for convex polyhedron
+template< class Shape >
+double IntegratorHPMCMonoNEC< Shape >::sweepDistance(vec3<Scalar>& direction,
+                                                    double maxSweep,
+                                                    unsigned int i,
+                                                    int& next,
+                                                    int typ_i,
+                                                    vec3<Scalar>& pos_i,
+                                                    Scalar4 postype_i,
+                                                    ShapeConvexPolyhedron& shape_i,
+                                                    ArrayHandle<unsigned int>& h_overlaps,
+                                                    ArrayHandle<Scalar4>& h_postype,
+                                                    ArrayHandle<Scalar4>& h_orientation,
+                                                    hpmc_nec_counters_t& nec_counters,
+                                                    vec3<Scalar>& collisionPlaneVector
+                                                   )
+    {
+    double sweepableDistance = maxSweep;
+
```

Comment:
Why is the normalisation of the `direction` missing here compared to the `sweepDistance` function using spheres?

Reply:
Simplifying the code with a `sweep_distance` for spheres, that difference does not exist anymore.